autorest/azure/cli/token.go

```diff
@@ -170,7 +170,11 @@ func GetTokenFromCLIWithParams(params GetAccessTokenParams) (*Token, error) {
 
 	output, err := cliCmd.Output()
 	if err != nil {
-		return nil, fmt.Errorf("Invoking Azure CLI failed with the following error: %s", stderr.String())
+		if _, ok := err.(*exec.ExitError); ok {
```

Comment:
Better to use `errors.As()` as that's the preferred pattern.

Another option, maybe it's simplest to check if `stderr.String()` for the empty string?

Reply:
Yeah, you are right a check if stderr is empty is the simplest. Updated the PR accordingly.